tools/realsense-viewer/auto_updater/update_handler.cpp

```diff
@@ -0,0 +1,271 @@
+        {
+            if (load_server_versions())
+            {
+                _server_versions_loaded = true;
+            }
+            else
+            {
+                return 0; // Cannot download server file
+            }
+        }
+
+        auto platform = get_platform();
+
+        // Look for the required version
+        auto res = std::find_if(_server_versions_vec.begin(), _server_versions_vec.end(),
+            [cp, up, platform](server_version_type ver) {return ((ver.component == cp) && (ver.policy == up) && (ver.platform == platform)); });
+
+        if (res != _server_versions_vec.end())
+        {
+            return (*res).version;
+        }
+
+        return 0; // Nothing found
+    }
+
+    bool update_handler::get_download_link(const component_part_type cp, const long long version, std::string& dl_link)
+    {
+        // Load server versions info on first access
+        if (!_server_versions_loaded)
+        {
+            if (load_server_versions())
+            {
+                _server_versions_loaded = true;
+            }
+            else
+            {
+                return 0; // Cannot download server file
+            }
+        }
+
+        std::string platform = get_platform();
+
+        // Look for the required version
+        auto res = std::find_if(_server_versions_vec.begin(), _server_versions_vec.end(),
+            [cp, version, platform](server_version_type ver) {return ((ver.component == cp) && (ver.version == version) && (ver.platform == platform)); });
+
+        if (res != _server_versions_vec.end())
+        {
+            dl_link = (*res).link;
+            return true;
+        }
+
+        return false;
+    }
+
+    bool update_handler::convert_component_name(const std::string& comp, component_part_type &out) const
+    {
+        bool convert_ok(true);
+        if (comp == "LIBREALSENSE") out = LIBREALSENSE;
+        else if (comp == "VIEWER") out = VIEWER;
+        else if (comp == "FIRMWARE") out = FIRMWARE;
+        else convert_ok = false;
+
+        return convert_ok;
+    }
+
+    bool update_handler::convert_update_policy(const std::string& comp, update_policy_type &out) const
+    {
+        bool convert_ok(true);
+        if (comp == "EXPERIMENTAL") out = EXPERIMENTAL;
+        else if (comp == "RECOMMENDED") out = RECOMMENDED;
+        else if (comp == "REQUIRED") out = REQUIRED;
+        else convert_ok = false;
+
+        return convert_ok;
+    }
+
+    bool update_handler::get_server_data(std::stringstream &ver_data)
+    {
+        bool server_data_retrieved(false);
+
+        if (!_dev_info_url.empty())
+        {
+            http_downloader hd;
+            if (hd.download_to_stream(_dev_info_url, ver_data))
+            {
+                server_data_retrieved = true;
+            }
+        }
+        else
+        {
+            // Temporary TBR
+            //read from local file
+            std::ifstream file("rs_updater.json");
```

Comment:
just to clarify - if network connection is not identified we should not fall back to file

Reply:
This case was for an empty URL - a patch for working with a local file for now on development.
I will add an enum/bool to the Ctor thats clarify the input  { FROM_FILE, FROM_SERVER}.
and align the code to support it.